### PR TITLE
Add minio-server

### PR DESCRIPTION
**Pull request by kenodegard**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by dbast** [comment]
The problematic thing is that this is a static binary that includes compiled code from all this [dependencies](https://github.com/minio/minio/blob/master/go.sum)... so all the licences have to be potentially be included... have a look at e.g. this feedstock where this is done with the help of the go-licenses tool during building https://github.com/conda-forge/traefik-feedstock

**Comment by kenodegard** [comment]
@jaimergp thankfully this turned out to be pretty easy to compile

@dbast `go-licenses` does a pretty good job and for the dependencies it couldn't verify (45) I manually checked and included the license in the feedstock, let me know if you think we should handle this a different way

**Comment by dbast** [comment]
hm. the idea of go-licenses is not to add the result (=all the licenses to the feestock), but only to the resulting conda package... thats how also https://github.com/conda-forge/traefik-feedstock does it. that way updates are much easier to handle.

**Comment by dbast** [comment]
yay... nice :)

**Review by dbast — APPROVED** [review]

File: recipes/minio-server/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+build:
+  number: 0
+  script_env:
+    - GIT_COMMIT={{ commit }}
+    - GIT_TIME={{ tag }}
+
+requirements:
+  build:
+    - go >=1.16
```
[the remark this answers is not in the document]
Comment (kenodegard) [reply recipes/minio-server/meta.yaml:25]: @xhochy already tried this

`compiler('go')` results in, e.g., `go_osx-64` on macOS which is 4 years old and doesn't satisfy the `>=1.16` requirement: https://anaconda.org/conda-forge/go_osx-64

[the remark this answers is not in the document]
Comment (xhochy) [reply recipes/minio-server/meta.yaml:25]: Ah, yes. You need `{{ compiler('go_nocgo') }}`

[the remark this answers is not in the document]
Comment (kenodegard) [reply recipes/minio-server/meta.yaml:25]: `go_nocgo_osx-64` isn't a valid package on conda-forge: https://anaconda.org/conda-forge/go_nocgo_osx-64

[the remark this answers is not in the document]
Comment (kenodegard) [reply recipes/minio-server/meta.yaml:25]: never mind, its `go-nocgo`

[the remark this answers is not in the document]
Comment (kenodegard) [reply recipes/minio-server/meta.yaml]: fixed in https://github.com/conda-forge/staged-recipes/pull/23830/commits/a17f0fdeebbbca43bd30fda250d7ca132f93934f

**Comment by jaimergp** [comment]
For the osx-64, can you temporarily pin `mamba <=1.4.2` in the [macOS script](https://github.com/kenodegard/staged-recipes/blob/3fed68d08cf4507090b9d78908d1174e4e43ad0f/.scripts/run_osx_build.sh#L26)?

**Comment by kenodegard** [comment]
@conda-forge/staged-recipes, ready for review!

**Pull request merged**